x-pack/plugins/alerting_builtins/server/alert_types/es_query/alert_type.ts

```diff
@@ -0,0 +1,134 @@
+    defaultMessage: 'ES Query',
+  });
+
+  const actionGroupName = i18n.translate(
+    'xpack.alertingBuiltins.esQuery.actionGroupQueryMatchedTitle',
+    {
+      defaultMessage: 'Query Matched',
+    }
+  );
+
+  return {
+    id: ID,
+    name: alertTypeName,
+    actionGroups: [{ id: ActionGroupId, name: actionGroupName }],
+    defaultActionGroupId: ActionGroupId,
+    executor,
+  };
+
+  async function executor({ params, services }: AlertExecutorOptions) {
+    const { query, filters = [], timefilter } = params as Params;
+    const index = find(filters, filter => !!filter.meta?.index)?.meta?.index;
+
+    const indexPattern = index
+      ? await findIndexPatternById(services.indexPattern, index)
+      : undefined;
+
+    const {
+      rawResponse: { hits: { total = 0, hits = [] } = {} },
+    } = (await services.search(
+      {
+        params: {
+          index,
+          body: {
+            query: buildEsQuery(
+              indexPattern,
+              query,
+              pipe(
+                fromNullable(timefilter),
+                mapNullable(timeFilter =>
+                  indexPattern ? getTime(indexPattern, timeFilter) : undefined
+                ),
+                fold(
+                  () => filters,
+                  timeFilter => [...filters, timeFilter]
+                )
+              )
+              // We'll need access to the UISettings via the *start* contract for this to work synchronously
+              // , await getEsQueryConfig(uiSettings)
+            ),
+          },
+        },
+      },
+      {},
+      'es'
+    )) as IEsSearchResponse;
+
+    hits.map(hit => {
+      services
+        .alertInstanceFactory(hit._id)
+        .replaceState(hit)
+        .scheduleActions('default');
+    });
+
+    return {
+      totalLastRun: total,
+    };
+  }
+}
+
+/** *
+ * We'll want to expose this as a service on the dat aplugin rather than go finding
+ * the SavedObject ourselves. This is here just to get a working prototype
+ */
+export const findIndexPatternById = async (
+  indexPattern: AlertServices['indexPattern'],
+  index: string
+): Promise<IIndexPattern | undefined> => {
+  try {
+    return await indexPattern.getById(index);
+  } catch (err) {
+    if (!SavedObjectsErrorHelpers.isNotFoundError(err)) {
+      throw err;
+    }
+  }
+  return;
+};
+
+// async function getEsQueryConfig(config: IUiSettingsClient) {
+//   const [
+//     allowLeadingWildcards,
+//     queryStringOptions,
+//     ignoreFilterIfFieldNotInIndex,
+//     dateFormatTZ,
+//   ] = await Promise.all([
+//     config.get('query:allowLeadingWildcards'),
+//     config.get('query:queryString:options'),
+//     config.get('courier:ignoreFilterIfFieldNotInIndex'),
+//     config.get('dateFormat:tz'),
+//   ]);
+
+//   return {
+//     allowLeadingWildcards,
+//     queryStringOptions,
+//     ignoreFilterIfFieldNotInIndex,
+//     dateFormatTZ,
+//   } as EsQueryConfig;
+// }
```

Comment:
We're going to need to also look at UISettings that dictate how the query is parsed in Discover, but Alerts are defined at `Setup` and we don't have access to these settings until `Start` se we'll need to play around with how this is bootstrapped.